src/main/java/org/sonarlint/intellij/core/SecurityHotspotMatcher.java

```diff
@@ -48,11 +48,17 @@ public LocalHotspot match(ServerHotspot serverHotspot) {
   }
 
   public Location matchLocation(ServerHotspot serverHotspot) {
-    for (VirtualFile contentRoot : ProjectRootManager.getInstance(project)
-      .getContentRoots()) {
-      VirtualFile matchedFile = contentRoot.findFileByRelativePath(serverHotspot.filePath);
-      if (matchedFile != null) {
-        return matchTextRange(matchedFile, serverHotspot.textRange, serverHotspot.message);
+    for (VirtualFile contentRoot : ProjectRootManager.getInstance(project).getContentRoots()) {
+      if (contentRoot.isDirectory()) {
+        VirtualFile matchedFile = contentRoot.findFileByRelativePath(serverHotspot.filePath);
+        if (matchedFile != null) {
+          return matchTextRange(matchedFile, serverHotspot.textRange, serverHotspot.message);
+        }
+      } else {
+        // On Rider, all source files are returned as individual content roots, so simply check for equality
+        if (contentRoot.getPath().endsWith(serverHotspot.filePath)) {
```

Comment:
Couldn't we match completely unrelated paths using `endsWith` ?

Reply:
Yep, it is possible, but quite unlikely, because this is the relative path from Solution baseDir, so having a mismatch would be really strange.